src/_pytest/python.py

```diff
@@ -911,6 +911,124 @@ def hasnew(obj: object) -> bool:
     return False
 
 
+@final
+@attr.s(frozen=True, auto_attribs=True)
+class IdMaker:
+    argnames: Iterable[str] = attr.Factory(list)
```

Comment:
Since only one place calls `IdMaker`, and it provides all parameters, I think we should get rid of the defaults of all fields.